<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro">
  <xacro:include filename="$(find tp_fanuc)/urdf/common_materials.xacro"/>

  <xacro:macro name="fanuc_lrmate200id4s" params="prefix">

    <!-- LINKS: main serial chain -->
    <link name="${prefix}base_link">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/base_link.stl"/>
        </geometry>
        <xacro:material_fanuc_gray28 />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/base_link.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_1">
      <visual>
        <origin xyz="0 0 0" rpy="${pi/2} 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_1.stl"/>
        </geometry>
        <xacro:material_fanuc_yellow />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_1.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_2">
      <visual>
        <origin xyz="-0.260 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_2.stl"/>
        </geometry>
        <xacro:material_fanuc_yellow />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_2.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_3">
      <visual>
        <origin xyz="-0.020 0 0" rpy="${pi/2} 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_3.stl"/>
        </geometry>
        <xacro:material_fanuc_yellow />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_3.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_4">
      <visual>
        <origin xyz="0 0 0" rpy="-${pi/2} 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_4.stl"/>
        </geometry>
        <xacro:material_fanuc_yellow />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_4.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_5">
      <visual>
        <origin xyz="0 0 0" rpy="${pi/2} 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_5.stl"/>
        </geometry>
        <xacro:material_fanuc_yellow />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_5.stl"/>
        </geometry>
      </collision>
    </link>

    <link name="${prefix}link_6">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/visual/link_6.stl"/>
        </geometry>
        <xacro:material_fanuc_gray28 />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://tp_fanuc/meshes/lrmate200id4s/collision/link_6.stl"/>
        </geometry>
      </collision>
    </link>



    <!-- JOINTS: main serial chain -->
    <joint name="${prefix}joint_1" type="revolute">
      <origin xyz="0 0 0.330" rpy="0 0 0" />
      <parent link="${prefix}base_link" />
      <child link="${prefix}link_1" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-2.965" upper="2.965" velocity="7.85" />
    </joint>

    <joint name="${prefix}joint_2" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="${prefix}link_1" />
      <child link="${prefix}link_2" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-1.745329" upper="2.530727" velocity="6.63" />
    </joint>

    <joint name="${prefix}joint_3" type="revolute">
      <origin xyz="0.260 0 0" rpy="0 0 0"/>
      <parent link="${prefix}link_2" />
      <child link="${prefix}link_3" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-1.745329" upper="2.530727" velocity="6.63" />
    </joint>

    <joint name="${prefix}joint_4" type="revolute">
      <origin xyz="0.020 0.260 0" rpy="0 0 0"/>
      <parent link="${prefix}link_3" />
      <child link="${prefix}link_4" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-1.745329" upper="2.530727" velocity="6.63" />
    </joint>

    <joint name="${prefix}joint_5" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="${prefix}link_4" />
      <child link="${prefix}link_5" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-1.745329" upper="2.530727" velocity="6.63" />
    </joint>

    <joint name="${prefix}joint_6" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="${prefix}link_5" />
      <child link="${prefix}link_6" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="-1.745329" upper="2.530727" velocity="6.63" />
    </joint>


  </xacro:macro>
</robot>
